src/server.tsx

```diff
@@ -42,7 +53,34 @@ app.get('/', async (req, res) => {
         console.log('Something went wrong: ', error.message);
         res.status(500).send({ error: error.message });
     }
-});
+};
+
+const metadataBuilder = (req: any): EpicMetadata => {
+    const {
+        ophanPageId,
+        ophanComponentId,
+        platformId,
+        campaignCode,
+        abTestName,
+        abTestVariant,
+        referrerUrl,
+    } = req.body;
+
+    return {
+        ophanPageId,
+        ophanComponentId,
+        platformId,
+        campaignCode,
+        abTestName,
+        abTestVariant,
+        referrerUrl,
+    };
+};
+
+const mockMetadataBuilder = (): EpicMetadata => mocks.metadata;
+
+app.post('/epic', epicHandler(metadataBuilder));
+app.get('/epic', epicHandler(mockMetadataBuilder));
```

Comment:
I think maybe this code is too clever for what it needs to do. Which makes things indirect and harder to modify. I.e. in contrast to a simpler:

```
app.get('/epic', async (req: any, res: any) => {
    const markup = buildEpic(testMetadata);
    res.send(renderHtmlDocument({ html, css }));
});

app.post('/epic', async (req: any, res: any) => {
    const metadata = metadataBuilder(req);
    const { html, css } = buildEpic(metadata);
    res.send({ html, css });
});
```
(This is not quite enough - need to also do error handling for example.)

But the idea is to still get code re-use (via functions) but avoid coupling the routes together by requiring branching - the `if (typeof req.query.showPreview !== 'undefined') {` part of the `epicHandler`. As branching of this sort makes things very brittle - i.e. easier to break preview when modifying the post behaviour and vice versa, and generally harder to understand because everything lives together so you need it to live in your head together too!

Reply:
I've refactored the code to use these principles - we treat the GET endpoint as preview and the POST endpoint for platform integration.

Reply:
Separately, I'll look into a middleware for error handling so we can further dry up those route handlers.